libs/community/langchain_google_community/vertex_rank.py

```diff
@@ -114,7 +117,11 @@ def _rerank_documents(
 
         records = [
             discoveryengine_v1alpha.RankingRecord(
-                id=str(idx),
+                id=(
+                    doc.metadata.get(self.id_field)
```

Comment:
should we maybe raise an error instead if the `id_field` is not in metadata?
or should we at least either always use enumerator OR id from metadata (i.e., if at least in one docs it's missing, let's use enumerator and raise a warning)?

Reply:
good idea. will add a check if `id_field` is missing: raise a `KeyError`

Reply:
added!